tutorials/intermediate/constrainedlayout_guide.py

```diff
@@ -260,66 +234,84 @@ def example_plot(ax, fontsize=12, hide_labels=False):
-# and the minimum distance between adjacent plots.  It is specified in
-# inches by the keyword arguments ``w_pad`` and ``h_pad`` to the function
-# `~.figure.Figure.set_constrained_layout_pads`:
+# Padding between axes is controlled in the horizontal by ``w_pad`` and
+# ``wspace``, and vertical by ``h_pad`` and ``hspace``.  These can be edited
+# via `~.figure.Figure.set_constrained_layout_pads`.  *w/h_pad* are
+# the minimum space around the axes in units of inches:
 
 fig, axs = plt.subplots(2, 2, constrained_layout=True)
 for ax in axs.flat:
     example_plot(ax, hide_labels=True)
-fig.set_constrained_layout_pads(w_pad=4/72, h_pad=4/72, hspace=0, wspace=0)
+fig.set_constrained_layout_pads(w_pad=4 / 72, h_pad=4 / 72, hspace=0, wspace=0)
+
+##########################################
+# Spacing between subplots is further set by ``wspace`` and ``hspace``. These
+# are specified as a fraction of the size of the subplot group as a whole.
+# If these values are smaller than w_pad or h_pad, then the fixed pads are used
+# instead. Note in the below how the space at the edges doesn't change from
+# the above, but the space between subplots does.
 
 fig, axs = plt.subplots(2, 2, constrained_layout=True)
 for ax in axs.flat:
     example_plot(ax, hide_labels=True)
-fig.set_constrained_layout_pads(w_pad=2/72, h_pad=2/72, hspace=0, wspace=0)
+fig.set_constrained_layout_pads(w_pad=4 / 72, h_pad=4 / 72, hspace=0.2,
+                                wspace=0.2)
 
 ##########################################
-# Spacing between subplots is set by ``wspace`` and ``hspace``. There are
-# specified as a fraction of the size of the subplot group as a whole.
-# If the size of the figure is changed, then these spaces change in
-# proportion.  Note in the below how the space at the edges doesn't change from
-# the above, but the space between subplots does.
+# If there are more than two columns, the wspace is shared between them,
+# so here the wspace is divided in 2, with a wspace of 0.1 between each column:
 
-fig, axs = plt.subplots(2, 2, constrained_layout=True)
+fig, axs = plt.subplots(2, 3, constrained_layout=True)
 for ax in axs.flat:
     example_plot(ax, hide_labels=True)
-fig.set_constrained_layout_pads(w_pad=2/72, h_pad=2/72, hspace=0.2, wspace=0.2)
+fig.set_constrained_layout_pads(w_pad=4 / 72, h_pad=4 / 72, hspace=0.2,
+                                wspace=0.2)
+
+##########################################
+# GridSpecs also have a ``hspace`` and ``wspace`` optional kwarg, that will
+# be used instead of the pads set by ``constrained_layout``:
 
+fig, axs = plt.subplots(2, 2, constrained_layout=True,
+                        gridspec_kw={'wspace': 0.3, 'hspace': 0.2})
+for ax in axs.flat:
+    example_plot(ax, hide_labels=True)
+# this has no effect because the space is set above:
+fig.set_constrained_layout_pads(w_pad=4 / 72, h_pad=4 / 72, hspace=0.0,
+                                wspace=0.0)
```

Comment:
`hspace` and `wspace` are gridspec properties, and I guess back when we did this the first time we decided to have them trump the constrained_layout supplied values.  Conversely, if someone had turned off constrained_layout, it would be odd to have the values set in CL affect the raw grid spec.  So, I'm open to changing this, but its been this way for quite a while...  I've updated the comment to be clearer...

Reply:
It doesn't have to change, but I think if you're explicitly passing `hspace`/`wspace` to `set_constrained_layout_pads`, then you expect it to do something, and it should warn if it's not going to.

Reply:
The problem is the ambiguity doesn't get resolved until draw time.  

I think the point here is that *most* people don't set `hspace` in gridspec, or even ever know there is such a thing as a gridspec - they just call `subplots` or `subplots_mosaic` etc.  However, you can easily imagine them thinking this should be something the layout manager would handle rather than pointing them to an implementation detail.  I think I'm OK with expert users overriding constrained_layouts hspace in the gridspec without warning.